### DISCUSSION TITLE
Backend Connection Issue in GitHub Codespaces Environment

### DISCUSSION THREAD

**Discussion by tictaqqn** [community/community#110862] [Codespaces]
### Select Topic Area

Bug

### Body

**Issue Summary:**
In our development environment, specifically using GitHub Codespaces, the frontend operates as expected when accessed through a URL ending in github.dev (replacing the traditional localhost:5174). Direct access to the backend, also via a github.dev URL (replacing localhost:3000), encounters no issues when accessed independently. However, attempts to connect to the backend from the frontend result in connection failure. This issue does not occur when the same setup is tested on a Mac environment with a traditional localhost setup, where the connection is established without any issues.

**Environment:**
- Backend: NestJS
- Frontend: Vite
- Node.js Version: 20
- Development Environment: GitHub Codespaces

**Steps to Reproduce:**
1. Start the backend on a github.dev URL (replacing the traditional localhost:3000).
2. Launch the frontend on a different github.dev URL (replacing the traditional localhost:5174).
3. Attempt to make API calls from the frontend to the backend.
4. The connection to the backend fails.

**Expected Result:**
Requests from the frontend should successfully reach the backend and be processed accordingly, despite the development environment being hosted on GitHub Codespaces.

**Actual Result:**
Connection attempts from the frontend to the backend fail, resulting in errors, suggesting a potential issue with the GitHub Codespaces networking or configuration setup.

**Attempted Solutions:**
- Restarting both backend and frontend services within Codespaces
- Verifying the version of Node.js
- Testing the setup in a Mac/WSL environment with traditional localhost (where it works without issues)

**Comment by erica-trump** [comment]
I'm having the same issue. On my end, it seems that the frontend requests are being redirected to a GitHub login page. I'm not sure how to resolve this. Did you make any progress with this @tictaqqn?

**Comment by sayhiben** [comment]
Try sending the `X-GitHub-Token` request header from your frontend. [Docs](https://docs.github.com/en/codespaces/developing-in-a-codespace/forwarding-ports-in-your-codespace#using-command-line-tools-and-rest-clients-to-access-ports)

**Comment by sajjadt87** [comment]
Hello I wanted to start programming what should I do first???

**Reply by queenofcorgis** [reply]
Welcome to GitHub and the GitHub Community @sajjadt87  🎆 ! We're glad you're here and can't wait to see where your GitHub journey takes you. 

Check out this [check-in](https://github.com/orgs/community/discussions/42302#discussion-4683140) with links on how to choose your first project to contribute to, a blog with getting started resources, and a new user checklist. We also have our [Programming Help category ](https://github.com/orgs/community/discussions/categories/programming-help) that may be useful as well. 

This post may not be the best spot for you as it is discussing a bug in a GitHub feature. You're more likely to get assistance when creating a post in the proper category and including all relevant details. 

Best of luck!

**Comment by bacaxnot** [comment]
Hi @tictaqqn did you solve this issue? I'm having the same problem

**Comment by 4Min4m** [comment]
I recently tackled a common issue in web development: CORS errors between a Vite + React frontend and a Node.js/Express backend in a GitHub Codespace. By configuring a proxy in vite.config.js, I was able to seamlessly connect the two without any CORS issues.

If you’re facing a similar challenge, check out my solution here:
https://github.com/4Min4m/proxyDemo.git
